### DISCUSSION TITLE
EspHome with S0 Interface eltako DSZ15D 3x80

### DISCUSSION THREAD

**Discussion by Sn4k3E4t3r** [klaasnicolaas/home-assistant-glow#208] [General]
Hi together,

i have an NodeMCU ESP8622 and i have no IR Led on my eltako dsz15d. But this Three-phase meter has an "S0+" an "S0-" Interface.

Has somebody the Three-phase meter  an read directly the Impulse from the S0 Interface

**Comment by ncvangilse** [comment]
Hey @Sn4k3E4t3r, did you ever find out? I am in same situation and reluctant to stick in a device in to the presumed S0 ports without being absolutely sure I will not fry myself (or the device)..

**Comment by Rikard74** [comment]
Hi @Sn4k3E4t3r @ncvangilse 
I have now tested this setup with S0. But i does not work as it is. Did you fint any solution?

**Reply by klaasnicolaas** [reply]
I removed your comment from the other thread because it doesn't belong there. 

What might work is to desolder the LED from the photodiode PCB and then a wire from the + and - in the S0 ports. But I have absolutely no experience with S0, a friend of mine made a nice project for it: https://github.com/huizebruin/s0tool

**Comment by steffenweber** [comment]
The instructions at https://esphome.io/cookbook/power_meter.html (see the note "Some energy meters have an exposed S0 port [...]") work for me using an Eltako DSZ15DE-3x80A attached to an ESP8266 (NodeMCU V2) running ESPHome.
